inst/tests/tests.Rraw

```diff
@@ -17264,12 +17271,40 @@ if (identical(x, enc2native(x))) {
 # fintersect now preserves order of first argument like intersect, #4716
 test(2163, fintersect(data.table(x=c("b", "c", "a")), data.table(x=c("a","c")))$x, c("c", "a"))
 
+
+# setDF now drops index attributes, #4889
+d <- data.table(a=1:100, b=1:100)
+setindex(d, a)
+setDF(d)
+d[1:50, "a"] <- d[51:100, "a"]
+setDT(d)
+test(2164.1, nrow(d[a==99]), as.integer(2))
+
+# In-place assignment can leak to origin DF, #4783
+ orig.opt <- options(datatable.assign.inplace=TRUE)
+ d <- data.frame(a=c(1,2), b=c(1,2))
+ df <- d
+ setDT(d)
+ d[, a:=c(10, 11)]  # not leaking to df
+ test(2164.2, df$a[1], 1)
+ d[!is.na(a), b:=c(20,21)]  # leaks to df
+ test(2166, df$b[1], 20)
```

Comment:
the issue is test numbering. here is 2166, next is 2164.3 -- tests should be in increasing order in the file.

Reply:
Thanks! I just added another commit (with other improvements) and hopefully this is fixed.